### DISCUSSION TITLE
Command ICommand working?

### DISCUSSION THREAD

**Discussion by WebAssemblyMan** [dotnet/maui#4068] [Q&A]
Hi,

Is Command (ICommand) is working in .NET MAUI Preview 11? I got stuck with a button click command as shown below. The savecommand does not trigger.

`
 <Button x:Name="SaveProductButton"
                Command="{Binding SaveCommand}"
                Text="Save Product"
                Grid.Row="6"
                Grid.Column="0"
                Grid.ColumnSpan="2"></Button>`

`

        public class ProductViewModel : INotifyPropertyChanged { 

        public ICommand SaveCommand { private set; get; } 
        public ProductViewModel() {
        SaveCommand = new Command(OnSaveCommand); 
        }
        public string UserName { get; set; }

        private void OnSaveCommand()
        {
            Console.WriteLine("Save Command");
        }

        public event PropertyChangedEventHandler PropertyChanged;

        public void RaisePropertyChanged(string propertyName)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }
    }
`

**Comment by nogginbox** [comment]
ICommand is working fine for me. I've used a slightly different syntax to hook it up, but I don't see anything wrong with the way you've done it.

This is what I have done:

```
public ICommand OnStartScan => new Command(() => StartScan(settings));
```
